src/lib/components/table/table-column-settings/table-settings.component.ts

```diff
@@ -59,16 +64,17 @@ export class TableSettingsComponent
                 width:        'auto',
                 disableClose: true,
                 data:         {
-                    columns:         this.table._contentColumnDefs,
+                    columns:         this.internalColumns ? this.internalColumns : this.table._contentColumnDefs.toArray(),
                     selectedColumns: this.selectedColumns
                 }
             });
 
-        dialogRef.afterClosed().subscribe((result:Array<string>) =>
+        dialogRef.afterClosed().subscribe((result:{ sortedList:Array<string>, columns:Array<MatColumnDef> }) =>
         {
             if(result !== null)
             {
-                this.selectedColumns = result;
+                this.selectedColumns = result.sortedList;
+                this.internalColumns = result.columns;
                 this.selectedColumnsChange.emit(this.selectedColumns);
             }
```

Comment:
Warum so umständlich?
Ich glaube internalColumns brauchst du nicht.
Nur die selectedColumns sind interessant.

Reply:
doch, ich muss es ja zwischenspeichern, damit es beim nächsten öffnen nicht der "initial" zustand kommt, sondern so wie man es eingestellt hat.

Reply:
Was für ein Initialzustand?
Die Liste hat eine reproduzierbare Sortierung, die immer wiederhergestellt werden kann.
Zuerst die ausgewählten Spalten, dann die nicht ausgewählten, in der Reihenfolge wie sie definiert wurden. Das brauchst du nicht speichern.
Wenn der Dialog geöffnet wird, rufst du ja die sortier Funktion auf.